src/mongocrypt-marking.c

```diff
@@ -271,6 +271,118 @@ DERIVE_TOKEN_IMPL(ESC)
 
 #undef DERIVE_TOKEN_IMPL
 
+/**
+ * Calculates:
+ * E?CToken = HMAC(collectionLevel1Token, n)
+ * E?CText<T>Token = HMAC(E?CToken, t)
+ * E?CText<T>DerivedFromDataTokenAndContentionFactorToken = HMAC(HMAC(E?CText<T>Token, v) cf)
+ *
+ * E?C = EDC|ESC|ECC
+ * n = 1 for EDC, 2 for ESC, 3 for ECC
+ * <T> = Exact|Substring|Suffix|Prefix
+ * t = 1 for Exact, 2 for Substring, 3 for Suffix, 4 for Prefix
+ * cf = contentionFactor
+ *
+ * E?CText<T>DerivedFromDataTokenAndContentionFactorToken is saved to out.
+ * Note that {out} is initialized even on failure.
```

Comment:
Why don't we cleanup on failure?

Reply:
"initialized" here just means we memset the buffer struct, not allocate memory, so there's nothing to clean up on failure.

Reply:
Makes sense, thanks.